float32 x
float32 y
float32 z
string name
string unique_name
string[] relations